vulcanai2/plotters/visualization.py

```diff
@@ -248,4 +246,31 @@ def display_record(record=None, load_path=None, interactive=True):
 #     sal_map = sal_fun(input_data)
 #     if sal_map.shape != input_data.shape:
 #         raise ValueError('Shape mismatch')
-#     return sal_map
\ No newline at end of file
+#     return sal_map
+
+def Saliency_map(image,model,preprocess,ground_truth,use_gpu=False,method=util.GradType.GUIDED):
```

Comment:
Saliency map should not be specific to just images. I realize that we only tested it for image based datasets but it will need to be generalized to any incoming datatype. Let's find a better way to send data to this folder. lets have this function signature start with (network, input_x, input_y,...). Im not sure what datatype the variable for `preprocess` is but I think it may stray away from the scope of this function. simply, we just have the input, put it through and get the back gradients, and return those. Let me know if you have any questions